app/models/shipment.rb

```diff
@@ -1,4 +1,21 @@
 class Shipment < ApplicationRecord
   belongs_to :company
-  belongs_to :shipment_item
+  has_many   :shipment_items
+
+  validates :destination_country, presence: true, length: { maximum: 3 }
+  validates :origin_country,      presence: true, length: { maximum: 3 }
+  validates :tracking_number,     presence: true
+  validates :slug,                presence: true
+
+  def group_and_order_items
+    private_group_items
+  end
+
+  private
+
+  def private_group_items(h=Hash.new(0))
+    shipment_items.each { |item| h[item.description] +=1 }
+    h.map { |k,v| {description: k,count: v} }.sort_by { |k| -k[:count] }
```

Comment:
You should use SQL query with `GROUP BY` since Database is more performant to deal with data than Ruby, right?

Few resources you can get some practices
- https://en.wikibooks.org/wiki/SQL_Exercises